Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/core/PersonCertificatesExtensions.kt

```diff
@@ -85,20 +85,34 @@ private fun Collection<CwaCovidCertificate>.rule2FindRecentRaCertificate(
  */
 private fun Collection<CwaCovidCertificate>.rule3FindRecentLastShot(
     nowUtc: Instant
-): CwaCovidCertificate? = this
-    .filterIsInstance<VaccinationCertificate>()
-    .filter {
-        with(it.rawCertificate.vaccination) { doseNumber == totalSeriesOfDoses }
+): CwaCovidCertificate? {
+    val twoDoseVaccines = listOf("EU/1/20/1528", "EU/1/21/1529", "EU/1/20/1507")
+    val isOlderThanTwoWeeks = { certificate: VaccinationCertificate ->
+        Days.daysBetween(
+            certificate.rawCertificate.vaccination.vaccinatedOn,
+            nowUtc.toLocalDateUtc()
+        ).days > 14
     }
-    .filter {
-        Days.daysBetween(it.rawCertificate.vaccination.vaccinatedOn, nowUtc.toLocalDateUtc()).days > 14
-    }
-    .maxWithOrNull(
-        compareBy(
-            { it.rawCertificate.vaccination.vaccinatedOn },
-            { it.headerIssuedAt }
+    return this
+        .filterIsInstance<VaccinationCertificate>()
+        .filter { with(it.rawCertificate.vaccination) { doseNumber == totalSeriesOfDoses } }
```

Comment:
I think you can use `vaccinationCertificate.isFinalShot`

Reply:
Thanks, didn't know that.